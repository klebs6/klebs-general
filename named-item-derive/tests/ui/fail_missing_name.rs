
// Should fail because there's no `name: String` field
// We expect the macro to error out.

use named_item_derive::NamedItem;
use named_item::{Named};

#[derive(NamedItem)]
struct MissingNameField {
    something_else: i32,
}

fn main() {}

