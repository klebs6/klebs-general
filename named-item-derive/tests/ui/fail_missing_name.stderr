error: Struct must have `name: String`.
 --> tests/ui/fail_missing_name.rs:9:8
  |
9 | struct MissingNameField {
  |        ^^^^^^^^^^^^^^^^

warning: unused import: `Named`
 --> tests/ui/fail_missing_name.rs:6:18
  |
6 | use named_item::{Named};
  |                  ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
